### DISCUSSION TITLE
Will full SPA in 3.x be optional?

### DISCUSSION THREAD

**Discussion by mikeseese** [requarks/wiki#5519] [Help / Questions]
I see on the [roadmap](https://docs.requarks.io/releases/roadmap) that full SPA (Single Page Application) will be added in 3.0. Is this going to be required/non-optional if we upgrade to 3.0? I'm interested in keeping my site SEO-friendly, and I believe SPA would hurt that quite a bit for users to find individual pages from search engines.

**Comment by NGPixel** [comment]
SPA doesn't prevent users from finding individual pages from search engines...

**Reply by 4e576rt8uh9ij9okp** [reply]
Hi there, doesn't that mean that the crawlers must run the javascript in order to index a page?
Most crawlers are simple HTML crawlers.

**Reply by NGPixel** [reply]
There's going to be server-side rendering of pages and a special render optimized for search engines. However, most crawlers can interpret pages with JS just fine. Saying they are just simple HTML crawlers is simply not true.
